### ISSUE TITLE
New "query" source doesn't accept tags

### ISSUE THREAD

**Issue by MrBloo63** [RafaelGB/obsidian-db-folder#207]
**Describe the bug**
When creating a database from "query" source, adding tags anywhere in "FROM ..." statement is identified as valid and yet triggers "Error obtaining query results". It seems as if "#" character from tag is identified as YAML comment and query is truncated. Trying to set query as escaped string from YAML configuration doesn't work since file seems to be overwritten on unidentified event.

**To Reproduce**
1. Open database settings
2. Select "query" as source
3. Enter "FROM #tag" as dataview query or any one including some tag
4. Validate query and close database settings
5. "Error obtaining query results" notification shows up

**Expected behavior**
Depending on target behavior, query should be identified as invalid or display filtered content. Target behavior is not clear for me since feature was introduced as a workaround for multiple source folders #89, which is working as expected. However behavior doesn't match screen recording from #156.

**Screenshots**
![Database Settings](https://user-images.githubusercontent.com/5083659/179475046-323b32f6-4966-4b4f-9a75-b3f29296049a.png)
![Error Notification](https://user-images.githubusercontent.com/5083659/179475481-74f23789-51a9-4f51-a47e-0fa262e5b087.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version: Obsidian 0.15.6 with 1.8.2 plugin (fresh vault and database)

**Comment by RafaelGB** [comment]
If the yaml inside database config is not wrapped with quotes, the `#values` are lost when is read

Could you share your markdown database file? This is strange. Should be wrap in all cases

**Comment by MrBloo63** [comment]
Thank you for your quick reply!

This were things are getting really weird. When setting query from GUI, YAML configuration shows `source_form_result: FROM`. If I switch to Markdown view and edit field as `source_form_result: "FROM #tag"`, and switch back to database view, I get the expected result. However, if I close file and open it again, I get "Error obtaining query results" once more. And Markdown view confirms that configuration has reverted to `source_form_result: FROM`. Closing and opening again Markdown view doesn't change content.

Does the database view rewrite YAML configuration when triggered?

For the record, I disabled all plugins except File Explorer, Dataview and DB Folder. I also disabled automatic brackets pairing and tried different languages. Should I check if it behaves the same on macOS? Or reinstall Obsidian from scratch (contrary to current update from v0.14.6)?
